templates/web-deployment.yaml

```diff
@@ -212,6 +216,16 @@ spec:
                   name: {{ template "concourse.web.fullname" . }}
                   key: local-users
             {{- end }}
+            - name: CONCOURSE_CLIENT_ID
```

Comment:
I think these will have to be somehow guarded by a condition like we do for the other uses of values right from secrets :thinking:  

at templating time, helm won't be able to tell if the contents of the value in the secret is filled or not, pretty much meaning that if not, we'll end up with the following environ vars:

```console
$ sudo cat /proc/$web_pid/environ | tr '\000' '\n' | grep CLIENT
CONCOURSE_CLIENT_ID=
CONCOURSE_CLIENT_SECRET=
CONCOURSE_TSA_CLIENT_ID=
CONCOURSE_TSA_CLIENT_SECRET=
```

which .. won't really work.

it's true that we could check if `.Values.secrets.clientId` is set, but we can't rely solely on that as we allow folks to bring their own `Secret` (instead of forcing people to always rely on the chart creating the Secret for them).

wdyt of having a `concourse.web.clientSecretConfig.enabled` or something like that?

```yaml
concourse:
  web:
    clientConfig:
       enabled: true
       id: blabla
       secret: balblablal
```

and then, the same for  `tsa`, etc?

Reply:
```
concourse:
  web:
    clientConfig:
       id: blabla
       secret: balblablal
```

Would we need the `id` and `secret` in this config? I think all the other cases where we have secrets defined we don't include it under the web params right?

So really it would just look like this?
```
concourse:
  web:
    clientConfig:
       enabled: true
```

Reply:
oh yeah, keeping `id` and `secret` under `secrets` :grin:  

> Which is really unfortunate because it's not very clear what you're enabling.

yeah :(

Reply:
I'm going to mark this conversation as resolved as it seems y'all implemented the change @cirocosta suggested. The force pushes make it confusing to read 😬